Table block: Make `figcaption` styles consistent between editor and front end

mikachan opened the pull request:
<!-- Thanks for contributing to Gutenberg! Please follow the Gutenberg Contributing Guidelines:
https://github.com/WordPress/gutenberg/blob/trunk/CONTRIBUTING.md -->

## What?
<!-- In a few words, what is the PR actually doing? -->
This PR makes the styling of the `figcaption` element in the Table block consistent between the editor and the front end. Currently, these are styled slightly differently.

## Why?
<!-- Why is this PR necessary? What problem is it solving? Reference any existing previous issue(s) or PR(s), but please add a short summary here, too -->
Visual consistency between the editor and the front end provides a better user experience. 

Also addresses a comment on trac here: https://core.trac.wordpress.org/ticket/56818#comment:16

## How?
<!-- How is your PR addressing the issue at hand? What are the implementation details? -->
This moves the styling for the table `figcaption` element to the block's `style.scss` file, rather than in the `editor.scss` and `theme.scss` files.

## Testing Instructions
<!-- Please include step by step instructions on how to test this PR. -->
<!-- 1. Open a Post or Page. -->
<!-- 2. Insert a Heading Block. -->
<!-- 3. etc. -->
1. Add a table block with a caption
2. Compare the styles between the editor and the front end

Example block markup:

```
<!-- wp:table {"align":"wide","className":"is-style-regular"} -->
<figure class="wp-block-table alignwide is-style-regular"><table><tbody><tr><td>Title</td><td>Title 2</td></tr><tr><td>Test test test</td><td>Test test test</td></tr></tbody></table><figcaption class="wp-element-caption">Caption Text</figcaption></figure>
<!-- /wp:table -->
```

## Screenshots or screencast <!-- if applicable -->

Before:
| Front end | Editor |
| --------- | ------- |
| <img width="1218" alt="image" src="https://user-images.githubusercontent.com/1645628/204628233-a577e040-de37-4cd8-aec7-2e8646a70017.png"> | <img width="1021" alt="image" src="https://user-images.githubusercontent.com/1645628/204628498-7ec5ec2b-f0b5-4e9a-9099-bb5001abe768.png"> |

After:
| Front end | Editor |
| --------- | ------- |
| <img width="1218" alt="image" src="https://user-images.githubusercontent.com/1645628/204627334-2504505e-5e64-4a0e-b977-78db853390cf.png"> | <img width="1021" alt="image" src="https://user-images.githubusercontent.com/1645628/204628888-bbed243d-af3b-4c0b-b3bf-d79f99c1b998.png"> |

cc. @WordPress/block-themers

mikachan committed:
Move figcaption styles to style.scss

jffng commented on packages/block-library/src/table/style.scss:
```diff
@@ -18,6 +18,13 @@
 
+	// Match caption styles across front end and editor
+	figcaption {
+		@include caption-style-theme();
```
It seems there is a convention for mixins with the suffix `-theme` to only use them in the theme styles — is it an issue to create an inconsistency here?

scruffian replied:
I think so yes :)

mikachan replied:
Thanks both, that makes sense. I've removed the use of this mixin here and instead used specific styles. I went with:

```
	figcaption {
		color: inherit;
		font-size: max(13px, 80%);
		text-align: center;
	}
```

The only difference here is that the font size is set to a minimum of 13px, as this is the same size that was set in the mixin, but then I've also allowed it to be 80% of the font size set by the theme. I'm not sure if this is too opinionated, but I thought it would allow for a more theme-related font size.

mikachan committed:
Remove theme mixin from style.scss

mikachan committed:
Increase font size percentage

jffng commented:
Thanks for working on this. I'm a bit split. Part of me thinks we should just remove the `caption-style-theme` from editor.scss, and leave the mixin in the theme styles. This way, the editor and front-end match and we aren't adding any opinion to the caption by default.

mikachan committed:
Remove figcaption styles from style, keep them in theme

mikachan commented:
> Part of me thinks we should just remove the caption-style-theme from editor.scss, and leave the mixin in the theme styles.

Hmm yeah, that fixes the consistency issue and keeps the theme styles more opinionated (which I think is their intention). 

Also, if we start to go in a direction where the opinionated styles are moved to style.scss / the default style, the theme.scss styles / `wp-block-styles` will eventually become redundant.

I've removed the more opinionated styles from style.scss, and added the mixin back to theme.scss, so we can test that out. Thanks for the feedback!

Just to clarify for anyone following along, this is what I'm now seeing with https://github.com/WordPress/gutenberg/pull/46172/commits/6671cbab313b72d75fc40ea9f86c01e62bc77238:

| Front end | Editor |
| ------- | ----- |
| ![image](https://user-images.githubusercontent.com/1645628/204860885-b52de3aa-1c2e-4e90-8c45-cc8240831c8b.png) | <img alt="image" src="https://user-images.githubusercontent.com/1645628/204861028-aacb08d6-819e-4c56-9737-d72817a4ccfb.png"> |

jffng reviewed (approved):
Makes sense to me!

scruffian commented:
Thanks for working on this. Should we investigate making `figcaption` an element in Global Styles?

mikachan commented:
> Should we investigate making figcaption an element in Global Styles?

Yeah, that makes sense. They're used in quite a few blocks, I think it would be great to style them via Global Styles.

scruffian commented:
If you're willing to create a PR, i'm willing to review it :D

mikachan commented:
We already have a `caption` element in Global Styles, which includes `figcaption` (under specific classes, anyway):

```
'caption' => '.wp-element-caption, .wp-block-audio figcaption, .wp-block-embed figcaption, .wp-block-gallery figcaption, .wp-block-image figcaption, .wp-block-table figcaption, .wp-block-video figcaption',
``` 

Do you think we should break this out so we have a separate `figcaption` element? I'm thinking that in most cases, these two elements would be styled similarly, so probably benefit from being part of the same element.

scruffian commented:
I think they should be the same. In that case, is it possible to migrate the styles from CSS to block.json
